docs/2.guide/4.recipes/1.custom-routing.md

```diff
@@ -51,12 +53,14 @@ export default defineNuxtConfig({
 
       // remove routes
       function removePagesMatching (pattern: RegExp, pages: NuxtPage[] = []) {
-        const pagesToRemove = []
+        const pagesToRemove: NuxtPage[] = []
         for (const page of pages) {
-          if (pattern.test(page.file)) {
-            pagesToRemove.push(page)
-          } else {
-            removePagesMatching(pattern, page.children)
+          if (page.file !== undefined) {
+            if (pattern.test(page.file)) {
```

Comment:
I think better to do something like:
```suggestion
            if (page.file && pattern.test(page.file)) {
```

As we still want to run the else block, even if file is undefined

Reply:
This is a recursion in else; Can't believe I didn't notice yesterday. 😂